src/Microsoft.OpenApi/Models/OpenApiDocument.cs

```diff
@@ -665,47 +665,52 @@ public bool AddComponent<T>(string id, T componentToRegister)
             Utils.CheckArgumentNull(componentToRegister);
             Utils.CheckArgumentNullOrEmpty(id);
             Components ??= new();
+
+            static Dictionary<string, TValue> AddToDictionary<TValue>(Dictionary<string, TValue>? dict, string key, TValue value)
```

Comment:
I think this method should return a boolean value to indicate whether the component was added or not.